consensus/src/network_interface.rs

```diff
@@ -117,18 +120,49 @@ impl NewNetworkSender for ConsensusNetworkSender {
 }
 
 impl ConsensusNetworkSender {
+    /// Initialize a shared hashmap about connections metadata that is updated by the receiver.
+    pub fn initialize(&mut self, connections: Arc<RwLock<HashMap<PeerId, ProtocolIdSet>>>) {
+        self.peers_protocols = Some(connections);
```

Comment:
the naming here is rather unfortunate... and -- maybe peers_to_supported_protocols
also it feels janky that we're passing around an arc/lock when I thought we were taking actual connection updates in other contexts? (please forgive me if my memory is off)

Reply:
this `peers_protocols` I think is relatively new.  I agree, maybe even just `protocols` or `peer_info` or something would be more appropriate.

The whole `Arc<RwLock<>>` or `Arc<Mutex<>>` model is usually used as a mechanism to get around `mutable` references in which you can only have one (for futures and things).  We probably don't need a `RwLock` but I'd have to look closer